tools/building.py

```diff
@@ -336,7 +336,7 @@ def acorn_optimizer(filename, passes, extra_info=None, return_output=False):
   cmd = config.NODE_JS + [optimizer, filename] + passes
   # Keep JS code comments intact through the acorn optimization pass so that JSDoc comments
   # will be carried over to a later Closure run.
-  if settings.USE_CLOSURE_COMPILER:
+  if settings.USE_CLOSURE_COMPILER or not settings.MINIFY_WHITESPACE:
```

Comment:
Perhaps it makes sense to rename `MINIFY_WHITESPACE`? But `MINIFY_WHITESPACE_AND_COMMENTS` seems overly verbose. Perhaps worth just adding that as a comment on the setting, at least?

Reply:
This is an internal settings, so not externally visible.  I don't see any urgent need to rename it.